source/materials/MaterialsList.cc

```diff
@@ -15,53 +15,79 @@
 #include <G4Element.hh>
 #include <G4NistManager.hh>
 
+#include <vector>
+#include <utility> 
+
 using namespace nexus;
 using namespace CLHEP;
 
-
 namespace materials {
-  G4Material* GXe(G4double pressure, G4double temperature)
-  {
-    G4Material* mat = GXe_bydensity(GXeDensity(pressure),
-                                    temperature, pressure);
+    // Calculation of Compressibilty factor from https://next.ific.uv.es/DocDB/0011/001183/001/GasDensity_2021-04-23.pdf
+    G4double CompressibilityFactor(G4double pressure, G4double temperature, double R) {
+        double T_r = temperature/289.733;
+        double P_r = pressure/ (58.420*bar);
+        double B0 = 0.083 - (0.422 / pow(T_r, 1.6));       
+        double z = 1 + B0*(P_r/T_r);
+        return z;       
+    }
+
+    // Function to calculate gas density based on Xe isotopic composition
+    G4double CalculateGasDensityFromIsotopicComposition(G4double pressure, G4double temperature, const std::vector<std::pair<int, double>>& isotopicComposition) {
+        const double R = 8.314; // Ideal gas constant in J/(mol·K)
+        double average_molar_mass = 0.0; // in g/mol
+
+        for (const auto& iso : isotopicComposition) {
+            int mass_number = iso.first;
+            double percentage = iso.second/100;
+            double molar_mass = XenonMassPerMole(mass_number) / (g / mole);
+            average_molar_mass += molar_mass * percentage;
+        }
+
+        average_molar_mass /= 1000.0;
```

Comment:
Why is this divided by 1000? I would avoid magic numbers, if possible or I would write a comment on that.